src/thunderbots/software/ai/navigator/path_planning_navigator/path_planning_navigator.cpp

```diff
@@ -176,3 +183,54 @@ void PathPlanningNavigator::visit(const StopIntent &stop_intent)
     auto p            = std::make_unique<StopPrimitive>(stop_intent);
     current_primitive = std::move(p);
 }
+
+std::optional<Obstacle> PathPlanningNavigator::obstacleFromAvoidArea(AvoidArea avoid_area)
+{
+    Rectangle rectangle({0, 0}, {0, 0});
+    switch (avoid_area)
+    {
+        case AvoidArea::FRIENDLY_DEFENSE_AREA:
+            // We extend the friendly defense area back by several meters to prevent
+            // robots going around the back of the goal
+            rectangle =
+                Rectangle(world.field().friendlyDefenseArea().neCorner(),
+                          Point(-10, world.field().friendlyDefenseArea().seCorner().y()));
+            rectangle.expand(OBSTACLE_INFLATION_DIST);
+            return Obstacle(rectangle);
+        case AvoidArea::ENEMY_DEFENSE_AREA:
+            // We extend the enemy defense area back by several meters to prevent
+            // robots going around the back of the goal
+            rectangle =
+                Rectangle(world.field().enemyDefenseArea().nwCorner(),
+                          Point(10, world.field().enemyDefenseArea().swCorner().y()));
+            rectangle.expand(OBSTACLE_INFLATION_DIST);
+            return Obstacle(rectangle);
+        case AvoidArea::INFLATED_ENEMY_DEFENSE_AREA:
+            rectangle = world.field().enemyDefenseArea();
+            rectangle.expand(OBSTACLE_INFLATION_DIST + 0.2);
+            return Obstacle(rectangle);
+        case AvoidArea::CENTER_CIRCLE:
+            return Obstacle(world.field().centerPoint(),
+                            world.field().centreCircleRadius() + OBSTACLE_INFLATION_DIST,
+                            NUM_POINTS_IN_CIRCLE_POLY);
+        case AvoidArea::HALF_METER_AROUND_BALL:
+            return Obstacle(world.ball().position(), 0.5 + OBSTACLE_INFLATION_DIST,
```

Comment:
I'd prefer to have an extra small buffer here too, since approximating circles with polygons isn't exact and we can afford to be a bit extra safe in this case. An extra robot radius or `0.1` should be fine.

Reply:
Added